src/array_api_stubs/_draft/_types.py

```diff
@@ -51,6 +56,65 @@
 # to keep pyflakes happy. https://github.com/python/typeshed/issues/3556
 ellipsis = TypeVar("ellipsis")
 
+DefaultDataTypes = TypedDict(
```

Comment:
I'm not sure that you can use `TypedDict` like this. It's a type hint only for regular dictionaries, and the docs always use a class which inherits from `TypedDict`: https://docs.python.org/3/library/typing.html#typing.TypedDict.

Reply:
Do you have a preferred alternative? I was not sure the best approach for the sake of typing.

Reply:
I'd be inclined to go with a regular `dict[str: dtype]`. However, let's see first if someone else has a more informed opinion - I have never used `TypedDict` before, so I meant my "I'm not sure" - this could well be fine.

Reply:
It should be alright I think: the `TypedDict` docs do state that this function-style syntax is supported, though the more conventional class-style syntax is generally preferred (the main use case for the function-style is dictionary keys with, _e.g._, hyphens or other characters that would lead to invalid symbols). 

Actually, is there any reason for _not_ using the class-style syntax here?

Reply:
@BvB93 Some of the dictionary keys contain spaces (e.g., `real floating` and `complex floating`).

For the other two dictionaries, I used the function syntax for consistency. I am happy to update if folks have strong opinions here.

Reply:
As the functional syntax for `TypedDict` is supported and `DefaultDataTypes` contains keys which are not valid identifiers (`real floating`), I'll leave as is and mark this as resolved.

Reply:
> As the functional syntax for `TypedDict` is supported and `DefaultDataTypes` contains keys which are not valid identifiers (`real floating`), I'll leave as is and mark this as resolved.

Reopen as per my question below: https://github.com/data-apis/array-api/pull/689#discussion_r1449754371